### PR TITLE
issue #156 - prevent mob session with outstanding local commits

### PR DESCRIPTION
**Pull request by TrentonAdams**
Let me know if this makes sense as a first start to fixing this issue.

### CODE REVIEW

**Comment by TrentonAdams** [comment]
@simonharrer Oh, i see what you mean.  I think this would work, yes?

```bash
echo hello > blah; git add blah; git commit -m 'blah'
git rev-list --count HEAD...refs/remotes/origin/main
1
```

Then if non zero, there's a local commit that wasn't pushed.

**Comment by simonharrer** [comment]
Yeah, let's try that. Can you introduce a new method that basically only checks that?

**Comment by TrentonAdams** [comment]
I sure can.  I'll see if I can get to that today.

**Comment by TrentonAdams** [comment]
I'm not sure what you wanted to see for line wrapping if any.  Let me know if you want anything changed.

File: mob.go [hunk 1]
```diff
@@ -751,6 +751,17 @@ func hasUncommittedChanges() bool {
 	return !isNothingToCommit()
 }
 
+func hasUnpushedCommits(branch string, configuration Configuration) bool {
+	countOutput := silentgit(
+		"rev-list", "--count", "refs/heads/"+branch+"..."+"refs/remotes/"+configuration.RemoteName+"/"+branch,
+	)
+	unpushedCount, err := strconv.Atoi(strings.TrimSpace(countOutput))
+	if err != nil {
+		panic(err)
```
Comment (simonharrer) [inline mob.go:760]: an info log about the number of unclpushed commits found would be nice

also in the lastest commit in main the silentgit method now always trims.

**Comment by simonharrer** [comment]
Can you add a changelog entry, please?

**Comment by simonharrer** [comment]
Otherwise looks really great. I already want to say thanks for contributing and helping out making the tool better! :-)

Looking forward to see this enhancement in the next version.

File: mob.go (hunk 1 above)
Comment (TrentonAdams) [reply mob.go:760]: k, I'll rebase against upstream in a moment.

**Comment by TrentonAdams** [comment]
k, done.  I can rebase to a squashed commit now if you like.  Or if you do that as part of your normal merge that's fine too.

**Comment by TrentonAdams** [comment]
Also, haven't tried from the actual cli yet, just the tests, but I'm going to go figure out how to install locally.  Been a couple months since I learned go stuff. lol

**Comment by TrentonAdams** [comment]
Confirmed...
```
$ ~/go/bin/mob start
   git fetch origin --prune
   git merge FETCH_HEAD --ff-only
 > starting new mob session from origin/main
   git checkout -B mob/main origin/main
   git push --no-verify --set-upstream origin mob/main
 > there are 1 unpushed commits
 ERROR cannot start; unpushed changes on current branch must be pushed upstream
 > you are mob programming
 > on wip branch mob/main (base branch main)
```

**Comment by TrentonAdams** [comment]
Oh, how come it says "you are mob programming on wip branch...".  That seems like an unrelated bug.  Cause we haven't started mob programming if there's an error.

**Comment by simonharrer** [comment]
The other output is something different. It’s something I would like to have a got as well. So if start returns early due to an error it should not call mob status afterwards. Probably needs a separate issue/ticket.

**Pull request merged**